CorsixTH/Lua/app.lua

```diff
@@ -1849,19 +1851,23 @@ function App:isAudioEnabled()
   return TH.GetCompileOptions().audio
 end
 
+--! Generate information about user's system and the program
+--!return gen_date (string) the date and time information was generated
+--!return (compiled .. running) (string) system and program info
 function App:getSystemInfo()
+  local gen_date = os.date("%Y-%m-%d %H:%M:%S")
   local compile_opts = TH.GetCompileOptions()
   local comp_details = {}
   for key, value in pairs(compile_opts) do
     table.insert(comp_details, key .. ": " .. tostring(value))
   end
   table.sort(comp_details)
   local compiled = string.format("Compiled with %s\nSDL renderer: %s\n",
-    table.concat(comp_details, ", "), self.video:getRendererDetails())
+      table.concat(comp_details, ", "), self.video:getRendererDetails())
   local running = string.format("%s run with api version: %s, game version: %s, savegame version: %s\n",
-    compile_opts.jit or _VERSION, tostring(corsixth.require("api_version")),
-    self:getVersion(), tostring(SAVEGAME_VERSION))
-  return (compiled .. running)
+      compile_opts.jit or _VERSION, tostring(corsixth.require("api_version")),
+      self:getVersion(), tostring(SAVEGAME_VERSION))
+  return gen_date, compiled .. running
```

Comment:
Not clear that we should be mixing conventions like this for this case.

What I mean is: getSystemInfo is currently concatenating the exact string to output in the game log. Now it's doing part of that construction and relying on the caller to do the rest. What set off alarms for me was that it was no longer providing it's primary function in the first return value, which is the normal convention.

I understand you're trying to abstract the function from it's use. I think we should do one of three things:
1. give up pretending this is an abstract function and call it `gameLogHeader()` and include `Gamelog generated on %s\n` in the returned string.
2. Include the date string in the returned string in such a way that it need not be specific to a game log e.g:
```
2022-12-07 08:59:00
Compiled with ...
```
3. Call os.date at both call sites where you're currently specifying `Gamelog generated on`.

3 feels like the worst option, the duplication you already have with the string is a good smell.

Reply:
> 1. give up pretending this is an abstract function and call it `gameLogHeader()` and include `Gamelog generated on %s\n` in the returned string

Option 1 done